src/homework8/arraylist/ArrayList.java

```diff
@@ -0,0 +1,107 @@
+package homework8.arraylist;
+
+import java.util.Arrays;
+
+public class ArrayList<T> implements List<T> {
+    private int position = 0;
+    private T[] array;
+
+    public ArrayList() {
+        array = (T[]) new Object[10];
```

Comment:
magic number, better change it

Reply:
What is the magic number? Why an array cannot have 10 elements?

Reply:
I understood - I must create the constant.